Fully ignore unknown tags in ParseCommentTags

jpbetz opened the pull request:
Fix `generators.ParseCommentTags` to not fail on parsing on tags that it ignores anyway.

thockin commented:
> I've made this opt-in since a full unit test run demonstrated that we have code that depends on the existing behavior.

Is that VALID, GOOD code or just code that exercised the existing behavior?

yongruilin commented on pkg/generators/openapi_test.go:
```diff
@@ -2386,7 +2387,7 @@ func TestMarkerComments(t *testing.T) {
 			StringValue string
 
-			// +k8s:validation:maxitems=10
+			// +k8s:validation:maxItems=10
```
Wondering why this passed the test before?

jpbetz replied:
Because maxItems is never used to check anything in the test :(  If the test had tried to exceed maxItems, it would have been allowed.

I could just remove this validation rules to simplify the test case..

yongruilin replied:
hmmm.. I think it is because the `json.Unmarshal` is [not case-sensitive](https://pkg.go.dev/encoding/json#Unmarshal). So when unmarshal the `commentTags` here https://github.com/kubernetes/kube-openapi/blob/99599406b04fd8d4b08b519eef9461744667043d/pkg/generators/markers.go#L411 , `maxItems` and `maxitems` both are parsed.

thockin replied:
We should not rely on that, I think the JSON we use in k/k is not case-insensitve

yongruilin replied:
k/k is using https://github.com/kubernetes-sigs/json which is case sensitive

jpbetz replied:
I'll follow up on this in a separate issue.

jpbetz commented:
> > I've made this opt-in since a full unit test run demonstrated that we have code that depends on the existing behavior.
> 
> Is that VALID, GOOD code or just code that exercised the existing behavior?

I dug deeper.  One of the problem cases was for the `[` delimited strings Alex added support for with the multiline cel support, and the other was a check for empty string.  I've fixed this to handle both and dropped the options.

thockin commented on pkg/generators/markers.go:
```diff
@@ -61,6 +63,30 @@ func (c *CELTag) Validate() error {
 	return nil
 }
 
+func isKnownTagCommentKey(key string) bool {
```
Could you comment this - Would be nice to explain the expected "grammar" for a key here.

jpbetz replied:
Sure, added.

thockin replied:
not pushed?

jpbetz replied:
oops, pushed now

thockin commented:
Edit PR comment to remote words about options?

jpbetz commented:
> Edit PR comment to remote words about options?

Thanks. Yes, that's misleading. Fixed.

thockin reviewed (commented):
LGTM

thockin commented:
Thanks!

/lgtm
/approve
